quickwit/quickwit-index-management/src/garbage_collection.rs

```diff
@@ -340,19 +341,33 @@ async fn delete_splits_marked_for_deletion_several_indexes(
             }
         };
 
+        // We page through the list of splits to delete using a limit and a `search_after` trick.
+        // To detect if this is the last page, we check if the number of splits is less than the
+        // limit.
+        assert!(splits_metadata_to_delete.len() <= DELETE_SPLITS_BATCH_SIZE);
+        splits_to_delete_possibly_remaining =
+            splits_metadata_to_delete.len() == DELETE_SPLITS_BATCH_SIZE;
+
         // set split after which to search for the next loop
         let Some(last_split_metadata) = splits_metadata_to_delete.last() else {
             break;
         };
         list_splits_query = list_splits_query.after_split(last_split_metadata);
 
-        let num_splits_to_delete = splits_metadata_to_delete.len();
+        let mut splits_metadata_to_delete_per_index: HashMap<IndexUid, Vec<SplitMetadata>> =
+            HashMap::with_capacity(storages.len());
 
-        let splits_metadata_to_delete_per_index: HashMap<IndexUid, Vec<SplitMetadata>> =
-            splits_metadata_to_delete
-                .into_iter()
-                .map(|meta| (meta.index_uid.clone(), meta))
-                .into_group_map();
+        for meta in splits_metadata_to_delete {
+            if !storages.contains_key(&meta.index_uid) {
+                info!(index_uid=?meta.index_uid, "split not listed in storage map: skipping");
+                continue;
+            }
+            if let Some(splits) = splits_metadata_to_delete_per_index.get_mut(&meta.index_uid) {
+                splits.push(meta);
+            } else {
+                splits_metadata_to_delete_per_index.insert(meta.index_uid.clone(), vec![meta]);
```

Comment:
this can be expressed with the [entry](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry) api, which imo is easier to read

Reply:
Yeah. I was avoiding the needless allocation. I can remove that.